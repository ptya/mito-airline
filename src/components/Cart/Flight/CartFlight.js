import React from "react";
import PropTypes from "prop-types";

// utils
import { convertDate } from "utils/convertDate";

// local elements
import Wrapper from "./elements/Wrapper";
import Calendar from "./elements/Calendar";
import Airport from "./elements/Airport";

const Flight = props => {
  const { from, to, flight } = props;
  let { departure, arrival } = flight;
  departure = convertDate(departure);
  arrival = convertDate(arrival);

  return (
    <Wrapper>
      <Calendar>
        <span>{departure.shortMonth}</span>
        <span>{departure.day}</span>
      </Calendar>
      <Airport>
        <span>{from.shortName}</span>
        <span>{to.shortName}</span>
        <span>
          {departure.weekday} {departure.time} â€“ {arrival.time}
        </span>
      </Airport>
    </Wrapper>
  );
};

Flight.propTypes = {
  flight: PropTypes.object.isRequired,
  from: PropTypes.object.isRequired,
  to: PropTypes.object.isRequired
};

export default Flight;
