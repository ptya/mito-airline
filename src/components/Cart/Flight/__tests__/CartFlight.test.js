import React from "react";
import { render, cleanup } from "@testing-library/react";

// components
import CartFlight from "../CartFlight";

// mocks
import {
  stationOne,
  stationTwo,
  flight
} from "components/providers/__mocks__/mockValues";

afterEach(() => {
  cleanup();
  console.error.mockClear();
});

console.error = jest.fn();

test("Renders without errors with isSeparated=false and matches snapshot", () => {
  const { container } = render(
    <CartFlight
      from={stationOne}
      to={stationTwo}
      flight={flight}
      isSeparated={false}
    />
  );
  expect(container).toMatchSnapshot();
  expect(console.error).not.toHaveBeenCalled();
});

test("Renders without errors with isSeparated=true and matches snapshot", () => {
  const { container } = render(
    <CartFlight
      from={stationOne}
      to={stationTwo}
      flight={flight}
      isSeparated={true}
    />
  );
  expect(container).toMatchSnapshot();
  expect(console.error).not.toHaveBeenCalled();
});
