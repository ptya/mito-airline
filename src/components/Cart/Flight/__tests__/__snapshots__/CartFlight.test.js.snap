// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without errors with isSeparated=false and matches snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 15px;
  background: #FFF;
}

.c1 {
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 40px;
  height: 46px;
  border: 2px solid #e1e1e1;
  border-radius: 3px;
  margin: 0;
  text-transform: uppercase;
  font-size: 13px;
  font-weight: bold;
  line-height: 21px;
}

.c1 span:first-child {
  background: #f2f2f2;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  margin-left: 10px;
  font-size: 14px;
  font-weight: bold;
}

.c2 span:first-child:after {
  content: " –";
}

.c2 span:last-child {
  font-size: 10px;
}

<div>
  <div
    class="c0"
    data-testid="cart-flight"
  >
    <p
      class="c1"
    >
      <span>
        Jul
      </span>
      <span>
        12
      </span>
    </p>
    <p
      class="c2"
    >
      <span>
        Barcelona
      </span>
      <span>
        Budapest
      </span>
      <span>
        Fri
         
        10:30
         – 
        13:30
      </span>
    </p>
  </div>
</div>
`;

exports[`Renders without errors with isSeparated=true and matches snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 15px;
  background: #FFF;
  border-top: 1px dotted #c1c1c1;
}

.c0:before {
  content: "";
  position: absolute;
  top: -7px;
  right: 0;
  left: 0;
  margin: auto;
  border-top: 10px solid #FFF;
  border-left: 10px solid #FFF;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  box-shadow: 1px 1px #c1c1c1;
  width: 0;
}

.c1 {
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 40px;
  height: 46px;
  border: 2px solid #e1e1e1;
  border-radius: 3px;
  margin: 0;
  text-transform: uppercase;
  font-size: 13px;
  font-weight: bold;
  line-height: 21px;
}

.c1 span:first-child {
  background: #f2f2f2;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  margin-left: 10px;
  font-size: 14px;
  font-weight: bold;
}

.c2 span:first-child:after {
  content: " –";
}

.c2 span:last-child {
  font-size: 10px;
}

<div>
  <div
    class="c0"
    data-testid="cart-flight"
  >
    <p
      class="c1"
    >
      <span>
        Jul
      </span>
      <span>
        12
      </span>
    </p>
    <p
      class="c2"
    >
      <span>
        Barcelona
      </span>
      <span>
        Budapest
      </span>
      <span>
        Fri
         
        10:30
         – 
        13:30
      </span>
    </p>
  </div>
</div>
`;
